### ISSUE TITLE
Memory usage increase crazily on iOS when running with multiple FlutterViewControllers

### ISSUE THREAD

**Issue by z870248707** [flutter/flutter#77567]
Hello, 
I ran the example from https://github.com/flutter/samples/tree/master/add_to_app/multiple_flutters with Flutter 2, and I found the memory usage increase crazily, shown as below:

The column `Growth` shows the memory increment.
<img width="1002" alt="1" src="https://user-images.githubusercontent.com/3069003/110284775-39f9a100-801d-11eb-83bd-01c617a9d600.png">

From the allocation details, we can see the mainly increment is the allocation of IOSurface, which seems allocated by Metal layer:
<img width="942" alt="2" src="https://user-images.githubusercontent.com/3069003/110284780-3b2ace00-801d-11eb-85db-2a84c2c94a39.png">

How to reproduce:
Run the [multiple_flutters](https://github.com/flutter/samples/tree/master/add_to_app/multiple_flutters) example with Instruments, and click the `next` button again and again.

## Details

<details>
<summary> flutter doctor -v</summary>
[✓] Flutter (Channel stable, 2.0.0, on macOS 11.2 20D64 darwin-x64, locale zh-Hans-CN)
    • Flutter version 2.0.0
    • Framework revision 60bd88df91 (5 天前), 2021-03-03 09:13:17 -0800
    • Engine revision 40441def69
    • Dart version 2.12.0

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK
    • Platform android-30, build-tools 30.0.2
    • ANDROID_HOME
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.4, Build version 12D4e
    • CocoaPods version 1.10.0

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio
    • Android Studio at /Applications/Android Studio 4.2 Preview.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)

[✓] VS Code (version 1.53.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.19.0

[✓] Connected device (2 available)
    • iPhone7P (mobile)  • ios            • iOS 14.1
    • Chrome (web)      • chrome                                   • web-javascript • Google Chrome
      88.0.4324.192

• No issues found!

**Comment by xster** [comment]
Right, it's the same issue as #77621. Thanks for reporting. Let's merge them.
